### PR TITLE
Changes behavior when not printing to a tty

### PR DESCRIPTION
**Pull request by cgonyeo**
Successive updates to the download progress will not print over each other when not printing to a tty. Instead they will print to new lines.

I was also going to not draw the bar when not printing to a tty, but the `io.Writer` isn't known by ioprogress when the bar draw function is generated.

When invoked in `systemd-run` as shown in https://github.com/coreos/rkt/issues/1282, it now looks like:

```
Sep 15 12:03:48 falcon systemd[1]: Started /home/derek/go/src/github.com/coreos/rkt/rkt/./rkt --debug --insecure-skip-verify run docker://redis.
Sep 15 12:03:51 falcon rkt[26364]: rkt: fetching image from docker://redis
Sep 15 12:03:58 falcon rkt[26364]: Downloading ba249489d0b6: [                                    ] 0 B/37.2 MB
Sep 15 12:03:58 falcon rkt[26364]: Downloading ba249489d0b6: [                                    ] 3.59 KB/37.2 MB
Sep 15 12:03:59 falcon rkt[26364]: Downloading ba249489d0b6: [=                                   ] 1.54 MB/37.2 MB
Sep 15 12:03:59 falcon rkt[26364]: Downloading ba249489d0b6: [===                                 ] 3.37 MB/37.2 MB
Sep 15 12:04:00 falcon rkt[26364]: Downloading ba249489d0b6: [=====                               ] 5.43 MB/37.2 MB
Sep 15 12:04:00 falcon rkt[26364]: Downloading ba249489d0b6: [=======                             ] 7.45 MB/37.2 MB
Sep 15 12:04:01 falcon rkt[26364]: Downloading ba249489d0b6: [=========                           ] 9.69 MB/37.2 MB
Sep 15 12:04:01 falcon rkt[26364]: Downloading ba249489d0b6: [===========                         ] 11.4 MB/37.2 MB
Sep 15 12:04:02 falcon rkt[26364]: Downloading ba249489d0b6: [=============                       ] 13.6 MB/37.2 MB
Sep 15 12:04:02 falcon rkt[26364]: Downloading ba249489d0b6: [==============                      ] 15.2 MB/37.2 MB
Sep 15 12:04:03 falcon rkt[26364]: Downloading ba249489d0b6: [================                    ] 16.7 MB/37.2 MB
Sep 15 12:04:03 falcon rkt[26364]: Downloading ba249489d0b6: [=================                   ] 18.3 MB/37.2 MB
Sep 15 12:04:04 falcon rkt[26364]: Downloading ba249489d0b6: [===================                 ] 19.9 MB/37.2 MB
Sep 15 12:04:04 falcon rkt[26364]: Downloading ba249489d0b6: [====================                ] 21.4 MB/37.2 MB
Sep 15 12:04:05 falcon rkt[26364]: Downloading ba249489d0b6: [=====================               ] 22.1 MB/37.2 MB
Sep 15 12:04:05 falcon rkt[26364]: Downloading ba249489d0b6: [======================              ] 23.6 MB/37.2 MB
Sep 15 12:04:06 falcon rkt[26364]: Downloading ba249489d0b6: [========================            ] 24.8 MB/37.2 MB
Sep 15 12:04:06 falcon rkt[26364]: Downloading ba249489d0b6: [=========================           ] 26 MB/37.2 MB
Sep 15 12:04:07 falcon rkt[26364]: Downloading ba249489d0b6: [==========================          ] 27.4 MB/37.2 MB
Sep 15 12:04:07 falcon rkt[26364]: Downloading ba249489d0b6: [============================        ] 29 MB/37.2 MB
Sep 15 12:04:08 falcon rkt[26364]: Downloading ba249489d0b6: [=============================       ] 30.2 MB/37.2 MB
Sep 15 12:04:08 falcon rkt[26364]: Downloading ba249489d0b6: [==============================      ] 31.1 MB/37.2 MB
Sep 15 12:04:09 falcon rkt[26364]: Downloading ba249489d0b6: [==============================      ] 31.9 MB/37.2 MB
Sep 15 12:04:09 falcon rkt[26364]: Downloading ba249489d0b6: [===============================     ] 32.9 MB/37.2 MB
Sep 15 12:04:10 falcon rkt[26364]: Downloading ba249489d0b6: [================================    ] 34.1 MB/37.2 MB
Sep 15 12:04:10 falcon rkt[26364]: Downloading ba249489d0b6: [==================================  ] 35.2 MB/37.2 MB
Sep 15 12:04:11 falcon rkt[26364]: Downloading ba249489d0b6: [=================================== ] 36.4 MB/37.2 MB
Sep 15 12:04:11 falcon rkt[26364]: Downloading ba249489d0b6: [====================================] 37.2 MB/37.2 MB
Sep 15 12:04:15 falcon rkt[26364]: Downloading 19de96c112fc: [                                    ] 0 B/32 B
Sep 15 12:04:15 falcon rkt[26364]: Downloading 19de96c112fc: [====================================] 32 B/32 B
Sep 15 12:04:15 falcon rkt[26364]: Downloading d990a769a35e: [                                    ] 0 B/1.7 KB
Sep 15 12:04:15 falcon rkt[26364]: Downloading d990a769a35e: [====================================] 1.7 KB/1.7 KB
Sep 15 12:04:16 falcon rkt[26364]: Downloading 8656a511ce9c: [                                    ] 0 B/5.94 MB
Sep 15 12:04:16 falcon rkt[26364]: Downloading 8656a511ce9c: [                                    ] 16.4 KB/5.94 MB
Sep 15 12:04:17 falcon rkt[26364]: Downloading 8656a511ce9c: [=========                           ] 1.53 MB/5.94 MB
Sep 15 12:04:17 falcon rkt[26364]: Downloading 8656a511ce9c: [====================                ] 3.34 MB/5.94 MB
Sep 15 12:04:18 falcon rkt[26364]: Downloading 8656a511ce9c: [====================================] 5.94 MB/5.94 MB
Sep 15 12:04:19 falcon rkt[26364]: Downloading f7022ac152fb: [                                    ] 0 B/93.6 KB
Sep 15 12:04:19 falcon rkt[26364]: Downloading f7022ac152fb: [=                                   ] 3.59 KB/93.6 KB
Sep 15 12:04:19 falcon rkt[26364]: Downloading f7022ac152fb: [====================================] 93.6 KB/93.6 KB
Sep 15 12:04:19 falcon rkt[26364]: Downloading 8e84d9ce7554: [                                    ] 0 B/611 KB
Sep 15 12:04:19 falcon rkt[26364]: Downloading 8e84d9ce7554: [                                    ] 3.59 KB/611 KB
Sep 15 12:04:20 falcon rkt[26364]: Downloading 8e84d9ce7554: [====================================] 611 KB/611 KB
Sep 15 12:04:20 falcon rkt[26364]: Downloading c9e5dd2a9302: [                                    ] 0 B/32 B
Sep 15 12:04:20 falcon rkt[26364]: Downloading c9e5dd2a9302: [====================================] 32 B/32 B
Sep 15 12:04:21 falcon rkt[26364]: Downloading 27b967cdd519: [                                    ] 0 B/32 B
Sep 15 12:04:21 falcon rkt[26364]: Downloading 27b967cdd519: [====================================] 32 B/32 B
Sep 15 12:04:21 falcon rkt[26364]: Downloading 3024bf5093a1: [                                    ] 0 B/32 B
Sep 15 12:04:21 falcon rkt[26364]: Downloading 3024bf5093a1: [====================================] 32 B/32 B
```

### CODE REVIEW

**Comment by cgonyeo** [comment]
It also behaves the same as before when printing to a terminal.

File: draw.go [hunk 1]
```diff
@@ -50,7 +60,13 @@ func DrawTerminalf(w io.Writer, f DrawTextFormatFunc) DrawFunc {
 		}
 		maxLength = len(line)
 
-		_, err := fmt.Fprint(w, line+"\r")
+		var terminate string
```
Comment (vcaputo) [inline draw.go]: why not just:

```
terminate := "\n"
if !isTerminal(w) {
        terminate = "\r"
}
```

**Comment by vcaputo** [comment]
Your commit snagged the .draw.go.swp file btw.

**Comment by cgonyeo** [comment]
I just added a helper function, so you can tell the function generating the bar drawing function what you intend to send the output to.

If you're not sending the output to a tty then the bar will always be "".

Results in output like this when invoked via `systemd-run`.

```
Sep 15 13:49:56 falcon systemd[1]: Started /home/derek/go/src/github.com/coreos/rkt/rkt/./rkt --debug --insecure-skip-verify run docker://redis.
Sep 15 13:50:00 falcon rkt[30047]: rkt: fetching image from docker://redis
Sep 15 13:50:06 falcon rkt[30047]: Downloading ba249489d0b6:  0 B/37.2 MB
Sep 15 13:50:06 falcon rkt[30047]: Downloading ba249489d0b6:  3.59 KB/37.2 MB
Sep 15 13:50:07 falcon rkt[30047]: Downloading ba249489d0b6:  933 KB/37.2 MB
Sep 15 13:50:07 falcon rkt[30047]: Downloading ba249489d0b6:  2.77 MB/37.2 MB
Sep 15 13:50:08 falcon rkt[30047]: Downloading ba249489d0b6:  4.88 MB/37.2 MB
Sep 15 13:50:08 falcon rkt[30047]: Downloading ba249489d0b6:  7.07 MB/37.2 MB
Sep 15 13:50:09 falcon rkt[30047]: Downloading ba249489d0b6:  9.56 MB/37.2 MB
Sep 15 13:50:09 falcon rkt[30047]: Downloading ba249489d0b6:  11.7 MB/37.2 MB
Sep 15 13:50:10 falcon rkt[30047]: Downloading ba249489d0b6:  14.4 MB/37.2 MB
Sep 15 13:50:10 falcon rkt[30047]: Downloading ba249489d0b6:  16.8 MB/37.2 MB
Sep 15 13:50:11 falcon rkt[30047]: Downloading ba249489d0b6:  19 MB/37.2 MB
Sep 15 13:50:11 falcon rkt[30047]: Downloading ba249489d0b6:  21.7 MB/37.2 MB
Sep 15 13:50:12 falcon rkt[30047]: Downloading ba249489d0b6:  24.1 MB/37.2 MB
Sep 15 13:50:12 falcon rkt[30047]: Downloading ba249489d0b6:  26.4 MB/37.2 MB
Sep 15 13:50:13 falcon rkt[30047]: Downloading ba249489d0b6:  28.6 MB/37.2 MB
Sep 15 13:50:13 falcon rkt[30047]: Downloading ba249489d0b6:  31.1 MB/37.2 MB
Sep 15 13:50:14 falcon rkt[30047]: Downloading ba249489d0b6:  33.8 MB/37.2 MB
Sep 15 13:50:14 falcon rkt[30047]: Downloading ba249489d0b6:  35.9 MB/37.2 MB
Sep 15 13:50:14 falcon rkt[30047]: Downloading ba249489d0b6:  37.2 MB/37.2 MB
Sep 15 13:50:18 falcon rkt[30047]: Downloading 19de96c112fc:  0 B/32 B
Sep 15 13:50:18 falcon rkt[30047]: Downloading 19de96c112fc:  32 B/32 B
Sep 15 13:50:19 falcon rkt[30047]: Downloading d990a769a35e:  0 B/1.7 KB
Sep 15 13:50:19 falcon rkt[30047]: Downloading d990a769a35e:  1.7 KB/1.7 KB
Sep 15 13:50:20 falcon rkt[30047]: Downloading 8656a511ce9c:  0 B/5.94 MB
Sep 15 13:50:20 falcon rkt[30047]: Downloading 8656a511ce9c:  3.59 KB/5.94 MB
Sep 15 13:50:20 falcon rkt[30047]: Downloading 8656a511ce9c:  1.27 MB/5.94 MB
Sep 15 13:50:21 falcon rkt[30047]: Downloading 8656a511ce9c:  3.46 MB/5.94 MB
Sep 15 13:50:21 falcon rkt[30047]: Downloading 8656a511ce9c:  5.9 MB/5.94 MB
Sep 15 13:50:21 falcon rkt[30047]: Downloading 8656a511ce9c:  5.94 MB/5.94 MB
Sep 15 13:50:22 falcon rkt[30047]: Downloading f7022ac152fb:  0 B/93.6 KB
Sep 15 13:50:22 falcon rkt[30047]: Downloading f7022ac152fb:  3.59 KB/93.6 KB
Sep 15 13:50:22 falcon rkt[30047]: Downloading f7022ac152fb:  93.6 KB/93.6 KB
```

File: draw.go [hunk 2]
```diff
@@ -20,6 +22,14 @@ func init() {
 	defaultDrawFunc = DrawTerminal(os.Stdout)
 }
 
+// isTerminal returns True when stdout is going to a tty, and false otherwise.
```
Comment (jonboulle) [inline draw.go]: s/stdout/w/

File: test.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package main
```
Comment (jonboulle) [inline test.go]: empty test?

Comment (cgonyeo) [inline test.go]: Whoops, didn't mean to commit that

**Comment by jonboulle** [comment]
LGTM, merge it

**Pull request merged**